rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -122,14 +122,14 @@ def __next__(self):
                     output = torch.empty(
                         dimensions, dtype=getattr(torch, torch_dtype))
                     self.labels_tensor = torch.empty(
-                        self.labels_size, dtype=getattr(torch, torch_dtype))
+                        self.labels_size, dtype=getattr(torch, "int32"))
```

Comment:
the labels  data_type is hardcoded to be int32 here. But in get_one_hot_labels() it is copied as output_mem_type. Is there a case where they mismatch? It is generally not a good idea to hardcode the data_type

Reply:
The output_tensor_lists dtype can by of any datatype specified by the user
output_mem_type can be HOST / GPU.
Internally in rocAL, the data-type of the labels is always int. So we do a copy to the int32 type and later conversions are handled in python